docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,129 @@ components: DataGrid, XGrid
 
 # Data Grid - Editing
 
-<p class="description">Data Grid will provide full support to create, read, update, and delete operations (CRUD).</p>
+<p class="description">The data grid has built-in edit capabilities that you can customize to your needs.</p>
+
+## Cell editing
+
+Cell editing allows editing the value of one cell at a time.
+Set `editable` property of the column definition `GridColDef` object to `true` to allow editing cells of this column.
+
+```tsx
+<DataGrid columns={[{ field: 'name', editable: true }]} />
+```
+
+### Start editing
+
+If a cell is editable and has focus, any of the following interactions will start the edit mode:
+
+- A <kbd class="key">Enter</kbd> keydown
+- A <kbd class="key">Backspace</kbd> or <kbd class="key">Delete</kbd> keydown. It will also delete the value and stops the edit mode instantly.
+- A keydown of any printable key, for instance `a`, `E`, `0`, or `$`
+- A double click on the cell
+- A call to `apiRef.current.setCellMode()`.
+
+```
+ /**
+  * Set the cellMode of a cell.
+  * @param GridRowId
+  * @param string
+  * @param 'edit' | 'view'
+  */
+ setCellMode: (id: GridRowId, field: string, mode: GridCellMode) => void;
+```
+
+{{"demo": "pages/components/data-grid/editing/BasicEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Stop editing
+
+If a cell is in edit mode and has focus, any of the following interactions will stop the edit mode:
+
+- A <kbd class="key">Escape</kbd> keydown. It will also roll back changes done in the value of the cell.
+- A <kbd class="key">Tab</kbd> keydown. It will also save and goes to next cell on the same row.
+- A <kbd class="key">Enter</kbd> keydown. It will also save and goes to next cell on the same column.
+- A mousedown outside the cell
+
+### Control cell editability
+
+In addition to the `editable` flag on columns, control which cell is editable using `isCellEditable` prop.
+
+In this demo, the column `Name` is not editable. Only the rows with an even `Age` value are editable.
+The editable cells have a green background for better visibility.
+
+{{"demo": "pages/components/data-grid/editing/IsCellEditableGrid.js", "bg": "inline"}}
+
+### Control editing
+
+The `editRowsModel` prop lets you control the editing state.
+You can handle the `onEditRowModelChange` callback to control the `GridEditRowsModel` state.
+
+{{"demo": "pages/components/data-grid/editing/EditRowModelControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Column with valueGetter
+
+You can control the committed value when the edit move stops with the `onEditCellChangeCommitted` prop.
+This is especially interesting when using the `valueGetter` on the column definition.
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Client-side validation
```

Comment:
The UX on the validation is not flying.

1. There is little contrast between the validated and invalidated state. It's far from obvious to see the difference

<img width="251" alt="Screenshot 2021-04-14 at 12 03 52" src="https://user-images.githubusercontent.com/3165635/114693229-af8e1680-9d19-11eb-917c-da51890e85c7.png">

vs.

<img width="268" alt="Screenshot 2021-04-14 at 12 05 28" src="https://user-images.githubusercontent.com/3165635/114693281-bae14200-9d19-11eb-97bf-1cf0b7dbbb75.png">

2. It's undeniably wrong on dark mode

<img width="255" alt="Screenshot 2021-04-14 at 12 04 08" src="https://user-images.githubusercontent.com/3165635/114693328-c7659a80-9d19-11eb-9e14-372409191597.png">